### ISSUE TITLE
Database issues after upgrade

### ISSUE THREAD

**Issue by JCollinsMPS** [grokability/snipe-it#14612]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

Not sure if this is a bug or just something didn't go right. 

We wanted to update Snipe but we did not have it installed via git so we followed the instructions to change it to a git install and (we thought) successfully updated and installed. 

Unfortunately, we should have looked closer at things after but we had some problems. Every asset and existing user is still in the database. No images are missing and everything appears intact.

However, the LDAP sync stopped working shortly after this. When I run the LDAP sync via the users menu I get a 500 server error and it doesn't appear to sync any new users.

Running the command via PowerShell this is the error:

![image](https://github.com/snipe/snipe-it/assets/124394362/0e50b1bc-1217-4c6a-ab1a-9096c17a8c04)

Another oddity, when a backup is made I cannot locate it and it doesn't appear to exist. 

running **php artisan migrate** returns nothing to migrate.

Not sure what the best course of action is here short of possibly rebuilding Snipe again.

### Reproduction steps

1. Attempted LDAP sync, fails.
2. Attempted LDAP sync via PowerShell, errors out.
3. Rebooted server, no change.
...

### Expected behavior

LDAP Sync had been working without an issue prior to Git install and server update to version Version v6.3.3 - build 12903 (master).

### Snipe-IT Version

Version v6.3.3 - build 12903 (master)

### Operating System

Windows Server 2022 Datacenter

### Web Server

IIS

### PHP Version

8.2.0

### Operating System

Windows 10

### Browser

Google Chrome

### Error messages

```shell
php - m output:
[PHP Modules]
bcmath
calendar
Core
ctype
curl
date
dom
exif
fileinfo
filter
gd
hash
iconv
imap
json
ldap
libxml
mbstring
mysqli
mysqlnd
openssl
pcre
PDO
pdo_mysql
Phar
random
readline
Reflection
session
SimpleXML
sodium
SPL
standard
tokenizer
xml
xmlreader
xmlwriter
zip
zlib

[Zend Modules]

Debug Log:

[2024-04-17 13:30:35] production.ERROR: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'snipeit.user_groups' doesn't exist (SQL: insert into `users` (`password`, `activated`, `username`, `last_name`, `first_name`, `employee_num`, `email`, `ldap_import`, `updated_at`, `created_at`) values (*** NO PASSWORD ***, 1, 32XXXX, P, Isabella, , 32XXXX@ls.org, 1, 2024-04-17 13:30:35, 2024-04-17 13:30:35)) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'snipeit.user_groups' doesn't exist (SQL: insert into `users` (`password`, `activated`, `username`, `last_name`, `first_name`, `employee_num`, `email`, `ldap_import`, `updated_at`, `created_at`) values (*** NO PASSWORD ***, 1, 32015519, Pitz, Isabella, , 32015519@redhawks.marshallpublicschools.org, 1, 2024-04-17 13:30:35, 2024-04-17 13:30:35)) at C:\\inetpub\\wwwroot\\snipeit\\vendor\\laravel\\framework\\src\\Illuminate\\Database\\Connection.php:712)
Stack trace:
#0 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(501): PDOStatement->execute()
#1 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(705): Illuminate\Database\Connection->Illuminate\Database\{closure}()
#2 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(672): Illuminate\Database\Connection->runQueryCallback()
#3 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(490): Illuminate\Database\Connection->run()
#4 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(454): Illuminate\Database\Connection->statement()
#5 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Query\Processors\Processor.php(32): Illuminate\Database\Connection->insert()
#6 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php(3028): Illuminate\Database\Query\Processors\Processor->processInsertGetId()
#7 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php(1657): Illuminate\Database\Query\Builder->insertGetId()
#8 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(1188): Illuminate\Database\Eloquent\Builder->__call()
#9 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(1153): Illuminate\Database\Eloquent\Model->insertAndSetId()
#10 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(994): Illuminate\Database\Eloquent\Model->performInsert()
#11 C:\inetpub\wwwroot\snipeit\app\Console\Commands\LdapSync.php(398): Illuminate\Database\Eloquent\Model->save()
#12 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): App\Console\Commands\LdapSync->handle()
#13 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#14 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#15 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#16 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\Container.php(653): Illuminate\Container\BoundMethod::call()
#17 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Command.php(136): Illuminate\Container\Container->call()
#18 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Command\Command.php(298): Illuminate\Console\Command->execute()
#19 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Command.php(120): Symfony\Component\Console\Command\Command->run()
#20 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(1040): Illuminate\Console\Command->run()
#21 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(301): Symfony\Component\Console\Application->doRunCommand()
#22 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(171): Symfony\Component\Console\Application->doRun()
#23 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Application.php(94): Symfony\Component\Console\Application->run()
#24 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php(129): Illuminate\Console\Application->run()
#25 C:\inetpub\wwwroot\snipeit\artisan(33): Illuminate\Foundation\Console\Kernel->handle()
#26 {main}

Next Illuminate\Database\QueryException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'snipeit.user_groups' doesn't exist (SQL: insert into `users` (`password`, `activated`, `username`, `last_name`, `first_name`, `employee_num`, `email`, `ldap_import`, `updated_at`, `created_at`) values (*** NO PASSWORD ***, 1, 32XXXX, P, Isabella, , 32XXXX@ls.org, 1, 2024-04-17 13:30:35, 2024-04-17 13:30:35)) in C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php:712
Stack trace:
#0 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(672): Illuminate\Database\Connection->runQueryCallback()
#1 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(490): Illuminate\Database\Connection->run()
#2 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php(454): Illuminate\Database\Connection->statement()
#3 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Query\Processors\Processor.php(32): Illuminate\Database\Connection->insert()
#4 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Query\Builder.php(3028): Illuminate\Database\Query\Processors\Processor->processInsertGetId()
#5 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php(1657): Illuminate\Database\Query\Builder->insertGetId()
#6 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(1188): Illuminate\Database\Eloquent\Builder->__call()
#7 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(1153): Illuminate\Database\Eloquent\Model->insertAndSetId()
#8 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Model.php(994): Illuminate\Database\Eloquent\Model->performInsert()
#9 C:\inetpub\wwwroot\snipeit\app\Console\Commands\LdapSync.php(398): Illuminate\Database\Eloquent\Model->save()
#10 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(36): App\Console\Commands\LdapSync->handle()
#11 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\Util.php(40): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#12 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(93): Illuminate\Container\Util::unwrapIfClosure()
#13 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(35): Illuminate\Container\BoundMethod::callBoundMethod()
#14 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Container\Container.php(653): Illuminate\Container\BoundMethod::call()
#15 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Command.php(136): Illuminate\Container\Container->call()
#16 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Command\Command.php(298): Illuminate\Console\Command->execute()
#17 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Command.php(120): Symfony\Component\Console\Command\Command->run()
#18 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(1040): Illuminate\Console\Command->run()
#19 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(301): Symfony\Component\Console\Application->doRunCommand()
#20 C:\inetpub\wwwroot\snipeit\vendor\symfony\console\Application.php(171): Symfony\Component\Console\Application->doRun()
#21 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Console\Application.php(94): Symfony\Component\Console\Application->run()
#22 C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php(129): Illuminate\Console\Application->run()
#23 C:\inetpub\wwwroot\snipeit\artisan(33): Illuminate\Foundation\Console\Kernel->handle()
#24 {main}  

ERROR
[2024-04-17 13:30:35] production.ERROR: PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'snipeit.user_groups' doesn't exist in C:\inetpub\wwwroot\snipeit\vendor\laravel\framework\src\Illuminate\Database\Connection.php:501#13 C:\\inetpub\\wwwroot\\snipeit\\vendor\\laravel\\framework\\src\\Illuminate\\Console\\Command.php(120): Symfony\\Component\\Console\\Command\\Command->run()
#14 C:\\inetpub\\wwwroot\\snipeit\\vendor\\symfony\\console\\Application.php(1040): Illuminate\\Console\\Command->run()
#15 C:\\inetpub\\wwwroot\\snipeit\\vendor\\symfony\\console\\Application.php(301): Symfony\\Component\\Console\\Application->doRunCommand()
#16 C:\\inetpub\\wwwroot\\snipeit\\vendor\\symfony\\console\\Application.php(171): Symfony\\Component\\Console\\Application->doRun()
#17 C:\\inetpub\\wwwroot\\snipeit\\vendor\\laravel\\framework\\src\\Illuminate\\Console\\Application.php(94): Symfony\\Component\\Console\\Application->run()
#18 C:\\inetpub\\wwwroot\\snipeit\\vendor\\laravel\\framework\\src\\Illuminate\\Foundation\\Console\\Kernel.php(129): Illuminate\\Console\\Application->run()
#19 C:\\inetpub\\wwwroot\\snipeit\\artisan(33): Illuminate\\Foundation\\Console\\Kernel->handle()
#20 {main}
"}
```

### Additional context

Upgrade

**Comment by JCollinsMPS** [comment]
Yea, I have had no luck resolving this. It initially looked like the user_id column was missing from the users and user_groups table. Added the columns as necessary but the error persists. My conclusion is we will likely need to rebuild the Snipe server.

**Comment by snipe** [comment]
> It initially looked like the user_id column was missing from the users and user_groups table. Added the columns as necessary but the error persists. 

You should never ever have to manually add fields to the database (and in fact should not, as that can add problems later)

Did you modify the database manually in the past? 

How old is the original Snipe-IT?

**Comment by JCollinsMPS** [comment]
Never did it in the past no. And the server was maybe 2 years old tops. I ended up restoring from a snapshot anyways and did a backup. I decided to just do a clean reinstall of the server. After I restore the backup successfully I will update this thread. My hunch has been this had to do with switching to the Git install.

**Comment by snipe** [comment]
Switching to a git install shouldn't even require messing with the database at all though (unless you were moving it to another server altogether), and the users table has always had an `user_id` field.

**Comment by JCollinsMPS** [comment]
Right. I didn't think so either but the issues didn't come up until after that. I don't recall but composer gave me some issues when I did the switch. I thought I had it resolved as everything appeared okay after.

**Comment by snipe** [comment]
There's nothing that switching to a git install would do to the database though. Like, it just pulls down files, it never touches the database, and nothing in that process would literally drop table columns. :-/

**Comment by JCollinsMPS** [comment]
So decided to check a couple of things. It looks like folder permissions were removed after the changeover to git. I thought I had checked this but I needed to Re-add IUSR to the folders specified in the installation directions. After doing so EVERYTHING is working again.

LDAP is working as expected once again, everything seems kosher and no longer getting any 500 server errors anywhere.

Figures!
